### ISSUE TITLE
Laplace smoothing in Naive Bayes Learner

### ISSUE THREAD

**Issue by pkkarri** [accord-net/framework#255]
Hi,
     I'm trying to build a discrete classification system based on a Naive Bayes Learner. All my attributes/features take discrete values. But my test sample instances have values for some attributes which did not occur in my training sample. Please let me know if there is a way to handle occurrence of new values in test sample. 
     Is laplace smoothing included in the naive bayes implementation.??? This would help.

Regards,
Praveen Karri.

**Comment by cesarsouza** [comment]
Hi there!

I have added an example on how to use Laplace smoothing in the Naïve Bayes learning algorithm. 

https://github.com/accord-net/framework/blob/40d00ce3c0d775f39a74c5ce504a7ce032cc1847/Unit%20Tests/Accord.Tests.MachineLearning/Bayes/NaiveBayesTest.cs#L380

Basically, in order to enable the Laplace smoothing, all that is necessary is to set

```
// Enable the use of the Laplace rule
learning.Options.InnerOption.UseLaplaceRule = true;
```

After creating the `NaiveBayesLearning` learning algorithm.

Hope it helps!

Regards,
Cesar

**Comment by cesarsouza** [comment]
Fixed on release 3.3.0.
